lib/plugins/aws/invokeLocal/index.js

```diff
@@ -177,8 +189,145 @@ class AwsInvokeLocal {
         this.options.context);
     }
 
-    throw new this.serverless.classes
-      .Error('You can only invoke Node.js, Python, Java & Ruby functions locally.');
+    return this.invokeLocalDocker();
+  }
+
+
+  checkDockerImage() {
+    const runtime = this.options.functionObj.runtime
+      || this.serverless.service.provider.runtime
+      || 'nodejs4.3';
+
+    return new BbPromise((resolve, reject) => {
+      const docker = spawn('docker', ['images', '-q', `lambci/lambda:${runtime}`]);
+      let stdout = '';
+      docker.stdout.on('data', (buf) => { stdout += buf.toString(); });
+      docker.on('close', error => (error ? reject(error) : resolve(Boolean(stdout.trim()))));
+    });
+  }
+
+  pullDockerImage() {
+    const runtime = this.options.functionObj.runtime
+      || this.serverless.service.provider.runtime
+      || 'nodejs4.3';
+
+    this.serverless.cli.log('Downloading base docker image...');
+
+    return new BbPromise((resolve, reject) => {
+      const docker = spawn('docker', ['pull', `lambci/lambda:${runtime}`]);
+      docker.on('close', error => (error ? reject(error) : resolve()));
+    });
+  }
+
+  getLayerPaths() {
+    const layers = _.mapKeys(
+      this.serverless.service.layers,
+      (value, key) => this.provider.naming.getLambdaLayerLogicalId(key)
+    );
+
+    return Promise.all(
```

Comment:
Best would be to stick to one Promise implementation (`BbPromise` in this case).

Reason being that we once hunted down a race condition in the Serverless Variables implementation for days after figuring out that we used native Promises and Bluebird Promises in the same codebase and one implementation (probably Bluebird) was slightly faster.